pkg/controllers/user/helm/controller.go

```diff
@@ -373,3 +396,16 @@ func (l *Lifecycle) writeKubeConfig(obj *v3.App, tempDir string, remove bool) (s
 	}
 	return kubeConfigPath, nil
 }
+
+func (l *Lifecycle) isSystemProject(app *v3.App) (bool, error) {
+	project, err := l.projectLister.Get(l.ClusterName, app.Namespace)
+	if err != nil {
+		return false, err
+	}
+	return labels.Set(systemProjectLabel).AsSelector().Matches(labels.Set(project.Labels)), nil
+}
+
+func (l *Lifecycle) isSystemComponent(app *v3.App) bool {
```

Comment:
I understand what you are trying to do here, but the enclosing function name is quite confusing? Can I call this function with a different app to figure out if it's a system component? I think you should have some kind of annotation or a label on the app to make it a system component. If not, just pick a different function name, something like `isSystemMonitoringApp`.